source/slang/slang-check-decl.cpp

```diff
@@ -6916,6 +6945,111 @@ namespace Slang
         }
     }
 
+    void SemanticsDeclBasesVisitor::_searchMembersWithHigherVisibility(StructDecl* structDecl, const DeclVisibility& ctorVisibility, List<VarDeclBase*>& resultMembers)
+    {
+        auto findMembers = [&](StructDecl* structDecl)
+        {
+            for (auto varDeclRef : getMembersOfType<VarDeclBase>(getASTBuilder(), structDecl, MemberFilterStyle::Instance))
+            {
+                auto varDecl = varDeclRef.getDecl();
+                if (getDeclVisibility(varDecl) >= ctorVisibility)
+                {
+                    resultMembers.add(varDecl);
+                }
+            }
+        };
+
+        for (auto inheritanceMember : structDecl->getMembersOfType<InheritanceDecl>())
+        {
+            auto declRefType = as<DeclRefType>(inheritanceMember->base.type);
+            if (!declRefType)
+                continue;
+
+            auto structOfInheritance = as<StructDecl>(declRefType->getDeclRef().getDecl());
+            if (!structOfInheritance)
+                continue;
+
+            findMembers(structOfInheritance);
+        }
+
+        findMembers(structDecl);
+    }
+
+    // If a struct's member has:
+    // 1. an initialize expersion: Struct S {int a = 1;}; or
+    // 2. a default value: Struct T{}; Struct S {T t;}
+    //    Note, If a type is not default initializable, it doesn't have default value.
+    // it can be associated with default value expression in the constructor signature.
+    // This function helps to check whether either of those 2 conditions are met and create
+    // a default value for the parameter.
+    // It's totally fine that there is no default value for the parameter, in this case, user
+    // code has to provide the argument for this parameter.
+    static Expr* _getParamDefaultValue(SemanticsVisitor* visitor, VarDeclBase* varDecl)
+    {
+        // 1st condition is easy, we can just use the init expression as the default value.
+        if (varDecl->initExpr)
+        {
+            return varDecl->initExpr;
+        }
+
+        // For the 2nd condition, we need to check if the type is default initializable, if so,
+        // we can use the default constructor.
+        return constructDefaultInitExprForVar(visitor, varDecl);
```

Comment:
Feels more this should be named `constructDefaultInitExprForType`

Reply:
Changed the name in this PR.